src/basic/virt.c

```diff
@@ -180,8 +179,43 @@ static int detect_vm_dmi(void) {
                                 return dmi_vendor_table[j].id;
                         }
         }
-#endif
+        return VIRTUALIZATION_NONE;
+}
 
+static int detect_vm_dmi(void) {
+#if defined(__i386__) || defined(__x86_64__) || defined(__arm__) || defined(__aarch64__)
+
+        int r;
+        r = detect_vm_dmi_vendor();
+        if(r == VIRTUALIZATION_AMAZON) {
+                /* The DMI vendor tables in /sys/class/dmi/id don't help us distinguish between Amazon EC2
+                   virtual machines and bare-metal instances, so we need to look deeper.  The EC2 firmware
```

Comment:
Multi line comments in our code base are usually in the form:

```
/*
 *
 * .. */
 ```

Reply:
fixed